src/app/libraries/email_lib.php

```diff
@@ -5,33 +5,71 @@
+        $this->load_view($view_name, $data);
+        $content = ob_get_contents();
+        ob_end_clean();
+
+        return $content;
+    }
+
+    public function compose_mail($from_account, $from_name = null) {
         global $email_cfg;
 
-        $this->mailer = new PHPMailer();
-
-        $this->mailer->isSMTP();
-        $this->mailer->Host       = $email_cfg['server_host'];
-        $this->mailer->Hostname   = $email_cfg['server_domain'];
-        $this->mailer->SMTPAuth   = true;
-        $this->mailer->SMTPSecure = "tls";
-        $this->mailer->Port       = $email_cfg['server_port'];
-        $this->mailer->Username   = $email_cfg['username'];
-        $this->mailer->Password   = $email_cfg['password'];
-        $this->mailer->isHTML(true);
-        $this->mailer->Encoding   = 'quoted-printable';
-        $this->mailer->CharSet    = 'UTF-8';
+        $mailer = new PHPMailer();
+
+        if (!isset($email_cfg['server_host'])
+            || !isset($email_cfg['server_domain'])
+            || !isset($email_cfg['server_port'])
+            || !isset($email_cfg['accounts'])
+            || !isset($email_cfg['accounts'][$from_account])
+            || !isset($email_cfg['accounts'][$from_account]['username'])
+            || !isset($email_cfg['accounts'][$from_account]['password'])
+            || !isset($email_cfg['accounts'][$from_account]['email'])
+        ) {
+            return false;
+        }
+
+        $from = $email_cfg['accounts'][$from_account];
+        if (!empty($from_name)) {
+            $from['from_name'] = $from_name;
+        }
+
+        $mailer->isSMTP();
+        $mailer->Host       = $email_cfg['server_host'];
+        $mailer->Hostname   = $email_cfg['server_domain'];
+        $mailer->SMTPAuth   = true;
+        $mailer->SMTPSecure = "tls";
+        $mailer->Port       = $email_cfg['server_port'];
+        $mailer->Username   = $from['username'];
+        $mailer->Password   = $from['password'];
+        $mailer->isHTML(true);
+        $mailer->Encoding   = 'quoted-printable';
+        $mailer->CharSet    = 'UTF-8';
+
+        $mailer->setFrom($from['email'], $from['from_name']);
+
+        return $mailer;
     }
 
-    public function send_mail($email_data) {
+    public function set_html_view($mail, $html_view, $data) {
+        $mail->Body = $this->get_view_output($html_view, $data);
+    }
 
-        $this->mailer->ClearAddresses();
+    public function set_text_view($mail, $text_view, $data) {
+        $mail->AltBody = $this->get_view_output($text_view, $data);
+    }
 
-        $this->mailer->setFrom($email_data['from_email'], $email_data['from_name']);
-        $this->mailer->addAddress($email_data['to_email'], $email_data['to_name']);
-        $this->mailer->Subject = $email_data['subject'];
-        $this->mailer->Body    = $email_data['html_body'];
-        $this->mailer->AltBody = $email_data['alt_body'];
-        return $this->mailer->send();
+    public function send_mail($mail, $email_data) {
```

Comment:
Because we are talking about a single mail all the time. I'll change the `$mailer` to `$mail` in the previous function too.